sdk/python/kubeflow/storage_initializer/s3.py

```diff
@@ -48,10 +48,22 @@ def download_dataset(self):
             region_name=self.config.region_name,
         )
 
-        # Download the file
-        s3_client.download_file(
-            self.config.bucket_name,
-            self.config.file_key,
-            os.path.join(VOLUME_PATH_DATASET, self.config.file_key),
+        response = s3_client.list_objects_v2(
+            Bucket=self.config.bucket_name, Prefix=self.config.file_key
         )
-        print(f"File downloaded to: {VOLUME_PATH_DATASET}")
+        # Download the file
+        for obj in response.get("Contents", []):
```

Comment:
IIUC, this `list_objects_v2` API is a heavy load.
So, could you use `objects` API and `filter` function?: https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/s3/bucket/objects.html#filter

Reply:
done